--- imap-2006c1/Makefile.orig	2006-09-15 20:10:32.000000000 +0200
+++ imap-2006c1/Makefile	2006-11-21 11:30:48.549505000 +0100
@@ -85,6 +85,7 @@
 # lnx	Linux with traditional passwords and crypt() in the C library
 #	 (see lnp, sl4, sl5, and slx)
 # lnp	Linux with Pluggable Authentication Modules (PAM)
+# lnps	Linux with Pluggable Authentication Modules (PAM) (with shared libc-client)
 # lmd	Mandrake Linux
 # lrh	RedHat Linux 7.2 and later
 # lsu	SuSE Linux (same as lrh)
@@ -305,7 +306,7 @@
 
 # Note on SCO you may have to set LN to "ln".
 
-a32 a41 aix bs3 bsi d-g d54 do4 drs epx ga4 gas gh9 ghp ghs go5 gsc gsg gso gul h11 hpp hpx lnp lyn mct mnt neb nec nto nxt nx3 osf os4 ptx qnx sc5 sco sgi sg6 shp sl4 sl5 slx snx soc sol sos uw2: an
+a32 a41 aix bs3 bsi d-g d54 do4 drs epx ga4 gas gh9 ghp ghs go5 gsc gsg gso gul h11 hpp hpx lnp lnps lyn mct mnt neb nec nto nxt nx3 osf os4 ptx qnx sc5 sco sgi sg6 shp sl4 sl5 slx snx soc sol sos uw2: an
 	$(BUILD) BUILDTYPE=$@
 
 # If you use sv4, you may find that it works to move it to use the an process.
diff -ur imap-2004b.orig/src/dmail/Makefile imap-2004b/src/dmail/Makefile
--- imap-2004b.orig/src/dmail/Makefile	2002-11-19 01:43:31.000000000 +0100
+++ imap-2004b/src/dmail/Makefile	2005-01-11 00:18:13.000000000 +0100
@@ -18,14 +18,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 dmail: $(CCLIENTLIB) dmail.o
 	$(CC) $(CFLAGS) -o dmail dmail.o $(LDFLAGS)
diff -ur imap-2004b.orig/src/imapd/Makefile imap-2004b/src/imapd/Makefile
--- imap-2004b.orig/src/imapd/Makefile	2004-06-29 23:26:28.000000000 +0200
+++ imap-2004b/src/imapd/Makefile	2005-01-11 00:18:13.000000000 +0100
@@ -45,12 +45,12 @@
 # Get local definitions from c-client directory
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS` $(NSBD) $(ENBD) -DANOFILE=\"$(ANO)\" \
 	-DALERTFILE=\"$(ALERT)\" -DNNTPFILE=\"$(NNTP)\" \
 	-DUSERALERTFILE=\"$(USERALERT)\" -DSHUTDOWNFILE=\"$(SHUTDOWN)\"
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 all:	imapd
 
diff -ur imap-2004b.orig/src/ipopd/Makefile imap-2004b/src/ipopd/Makefile
--- imap-2004b.orig/src/ipopd/Makefile	2000-10-25 01:55:07.000000000 +0200
+++ imap-2004b/src/ipopd/Makefile	2005-01-11 00:18:13.000000000 +0100
@@ -19,14 +19,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 ipopd: ipop2d ipop3d
 
diff -ur imap-2004b.orig/src/mailutil/Makefile imap-2004b/src/mailutil/Makefile
--- imap-2004b.orig/src/mailutil/Makefile	2002-11-19 01:41:46.000000000 +0100
+++ imap-2004b/src/mailutil/Makefile	2005-01-11 00:18:13.000000000 +0100
@@ -19,14 +19,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 mailutil: $(CCLIENTLIB) mailutil.o
 	$(CC) $(CFLAGS) -o mailutil mailutil.o $(LDFLAGS)
diff -ur imap-2004b.orig/src/mtest/Makefile imap-2004b/src/mtest/Makefile
--- imap-2004b.orig/src/mtest/Makefile	2000-10-25 01:55:39.000000000 +0200
+++ imap-2004b/src/mtest/Makefile	2005-01-11 00:18:13.000000000 +0100
@@ -19,14 +19,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 all:	mtest
 
--- imap-2006j/src/osdep/unix/Makefile.orig	2007-07-12 19:01:17.428207961 +0200
+++ imap-2006j/src/osdep/unix/Makefile	2007-07-12 19:02:02.394770459 +0200
@@ -184,7 +184,7 @@
 	 SPOOLDIR=/var/spool \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -Dunix=1 -D_BSD" \
-	 BASELDFLAGS="-lbsd"
+	 BASELDFLAGS="../c-client/c-client.a -lbsd"
 
 a41:	# AIX 4.1 for RS/6000
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -200,12 +200,13 @@
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="-g" \
-	 BASELDFLAGS="-lbsd"
+	 BASELDFLAGS="../c-client/c-client.a -lbsd"
 
 aos:	# AOS for RT
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
-	 BASECFLAGS="-g -Dconst="
+	 BASECFLAGS="-g -Dconst=" \
+	 BASELDFLAGS="../c-client/c-client.a"
 
 art:	# AIX 2.2.1 for RT
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -214,7 +215,8 @@
 	 ACTIVEFILE=/usr/local/news/control/active \
 	 RSHPATH=/bin/rsh \
 	 BASECFLAGS="-g -Dconst= -Dvoid=char" \
-	 RANLIB=true 
+	 RANLIB=true \
+	 BASELDFLAGS="../c-client/c-client.a"
 
 asv:	# Altos SVR4
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -222,7 +224,7 @@
 	 ACTIVEFILE=/usr/spool/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-Dconst= -DSIGSTOP=SIGKILL" \
-	 BASELDFLAGS="-lsocket -lrpc -lgen -lcrypt -lxenix" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lrpc -lgen -lcrypt -lxenix" \
 	 RANLIB=true
 
 aux:	# A/UX
@@ -230,6 +232,7 @@
 	 CRXTYPE=nfs \
 	 MAILSPOOL=/usr/mail \
 	 BASECFLAGS="-g -B/usr/lib/big/ -Dvoid=char -Dconst=" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true ARRC="ar -rc"
 
 bs3:	# BSD/i386 3.0 or higher
@@ -238,11 +241,13 @@
 	 SPOOLDIR=/var NEWSSPOOL=/var/news/spool \
 	 ACTIVEFILE=/var/news/etc/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)" CC=shlicc
 
 bsd:	# BSD UNIX
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -Dconst="
 
 bsf:	# FreeBSD
@@ -260,6 +265,7 @@
 	 SPOOLDIR=/var NEWSSPOOL=/var/news/spool \
 	 ACTIVEFILE=/var/news/etc/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 bso:	# OpenBSD
@@ -268,6 +274,7 @@
 	 SPOOLDIR=/var \
 	 ACTIVEFILE=/usr/local/news/lib/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 cvx:	# Convex
@@ -293,7 +300,7 @@
 	 ACTIVEFILE=/local/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-lnsl -lsocket" \
+	 BASELDFLAGS="../c-client/c-client.a -lnsl -lsocket" \
 	 RANLIB=true
 
 d54:	# Data General DG/UX 5.4
@@ -303,7 +310,7 @@
 	 ACTIVEFILE=/local/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-lnsl -lsocket" \
+	 BASELDFLAGS="../c-client/c-client.a -lnsl -lsocket" \
 	 RANLIB=true
 
 dpx:	# Bull DPX/2
@@ -311,7 +318,7 @@
 	 SIGTYPE=sv4 CHECKPW=sv4 LOGINPW=sv4 \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-Dconst= -DSYSTEM5 -DSHORT_IDENT" \
-	 BASELDFLAGS="-linet" \
+	 BASELDFLAGS="../c-client/c-client.a -linet" \
 	 RANLIB=true LN=ln
 
 drs:	# ICL DRS/NX
@@ -321,18 +328,20 @@
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-O" \
-	 BASELDFLAGS="-lsocket -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lgen" \
 	 RANLIB=true
 
 do4:	# Apollo Domain/OS sr10.4
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="-A systype,bsd4.3 -D_APOLLO_SOURCE" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true
 
 dyn:	# Dynix
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -Dconst="
 
 epx:	# EP/IX
@@ -351,7 +360,7 @@
 	 SPOOLDIR=/var/spool \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -Dunix=1 -D_BSD" \
-	 BASELDFLAGS="-ls"
+	 BASELDFLAGS="../c-client/c-client.a -ls"
 
 gas:	# GCC Altos SVR4
 	$(BUILD) `$(CAT) SPECIALS` OS=asv \
@@ -359,7 +368,7 @@
 	 ACTIVEFILE=/usr/spool/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-g -O -DALTOS_SYSTEM_V -DSIGSTOP=SIGKILL" \
-	 BASELDFLAGS="-lsocket -lrpc -lgen -lcrypt -lxenix" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lrpc -lgen -lcrypt -lxenix" \
 	 RANLIB=true CC=gcc
 
 gh9:	# GCC HP-UX 9.x
@@ -368,6 +377,7 @@
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true CC=gcc
 
 ghp:	# GCC HP-UX 10.x
@@ -377,6 +387,7 @@
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true CC=gcc
 
 ghs:	# GCC HP-UX with Trusted Computer Base
@@ -386,7 +397,7 @@
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lnet -lV3 -lsec" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3 -lsec" \
 	 RANLIB=true CC=gcc
 
 go5:	# GCC 2.7.1 (95q4) SCO Open Server 5.0.x
@@ -397,7 +408,7 @@
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="$(GCCCFLAGS) -I/usr/include -L/lib" \
-	 BASELDFLAGS="-lsocket -lprot -lx -ltinfo -lm" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lx -ltinfo -lm" \
 	 RANLIB=true CC=gcc
 
 gsc:	# Santa Cruz Operation
@@ -406,7 +417,7 @@
 	 CREATEPROTO=mmdfproto EMPTYPROTO=mmdfproto \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lsocket -lprot -lcrypt_i -lx -los" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lcrypt_i -lx -los" \
 	 RANLIB=true LN=ln CC=gcc
 
 gsg:	# GCC Silicon Graphics
@@ -415,6 +426,7 @@
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bsd/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true CC=gcc
 
 gso:	os_sol.h	# GCC Solaris
@@ -424,21 +436,21 @@
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true CC=gcc
 
 gsu:	# GCC SUN-OS
 	$(BUILD) `$(CAT) SPECIALS` OS=sun \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-ldl" \
+	 BASELDFLAGS="../c-client/c-client.a -ldl" \
 	 CC=gcc
 
 gul:	# GCC Ultrix
 	$(BUILD) `$(CAT) SPECIALS` OS=ult \
 	 SIGTYPE=psx CHECKPW=ult CRXTYPE=nfs \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lauth -lc" \
+	 BASELDFLAGS="../c-client/c-client.a -lauth -lc" \
 	 CC=gcc
 
 h11:	# HP-UX 11i
@@ -456,7 +468,7 @@
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Aa -D_HPUX_SOURCE" \
-	 BASELDFLAGS="-lnet -lV3" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3" \
 	 RANLIB=true
 
 hpx:	# HP-UX 10.x
@@ -466,7 +478,7 @@
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Ae" \
-	 BASELDFLAGS="-lnet -lV3" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3" \
 	 RANLIB=true
 
 isc:	# Interactive
@@ -475,9 +487,21 @@
 	 SPOOLDIR=/var/spool MAILSPOOL=/var/mail \
 	 ACTIVEFILE=/var/spool/news/active \
 	 BASECFLAGS="-Xp -D_SYSV3" \
-	 BASELDFLAGS="-linet -lnsl_s -lgen -lx -lsec -liberty" \
+	 BASELDFLAGS="../c-client/c-client.a -linet -lnsl_s -lgen -lx -lsec -liberty" \
 	 RANLIB=true
 
+lnps:  # Linux Pluggable Authentication modules (c-client as shared lib)
+	$(BUILD) `cat EXTRASPECIALS` OS=lnx \
+	SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
+	SPOOLDIR=/var/spool \
+	ACTIVEFILE=/var/lib/news/active \
+	RSHPATH=/usr/bin/rsh \
+	BASECFLAGS="$(GCCCFLAGS)" \
+	ARCHIVE="libc-client.so" \
+	BASELDFLAGS="$(LDOPT) -L../c-client -lc-client -lpam -ldl -lc" \
+	ARRC="gcc --shared -Wl,-soname,libc-client.so.$(VERSION) -lssl -lcrypto -lpam -ldl -lc -o" \
+	RANLIB=true
+
 lnp:	# Linux Pluggable Authentication modules
 	$(BUILD) `$(CAT) SPECIALS` OS=slx \
 	 SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
@@ -485,7 +509,7 @@
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="$(PAMLDFLAGS)"
+	 BASELDFLAGS="../c-client/c-client.a $(PAMLDFLAGS)"
 
 lnx:	# Linux non-shadow passwords
 	@echo You are building for traditional Linux *without* shadow
@@ -504,12 +528,13 @@
 	 CRXTYPE=nfs \
 	 RSHPATH=/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS=-lbsd \
+	 BASELDFLAGS="../c-client/c-client.a -lbsd" \
 	 CC=gcc
 
 mct:	# MachTen - CRXTYPE=nfs doesn't work (at least not on 2.2)
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SPOOLDIR=/var/spool \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 mnt:	# Mint
@@ -527,7 +552,7 @@
 	 ACTIVEFILE=/var/db/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lcrypt"
+	 BASELDFLAGS="../c-client/c-client.a -lcrypt"
 
 nec:	# NEC UX
 	$(BUILD) `$(CAT) SPECIALS` OS=sv4 \
@@ -536,7 +561,7 @@
 	 ACTIVEFILE=/var/news/lib/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -Kopt=2 -KOlimit=2000" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true CC=/usr/abiccs/bin/cc
 
 nto:	# QNX Neutrino RTP
@@ -545,16 +570,19 @@
 	 SPOOLDIR=/var/spool \
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -O"
 
 nxt:	# NEXTSTEP
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 nx3:	# NEXTSTEP 3.0 single threaded
 	$(BUILD) `$(CAT) SPECIALS` OS=nxt \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 	echo "void malloc_singlethreaded (void);" >> linkage.h
 	echo "  malloc_singlethreaded ();" >> linkage.c
@@ -563,6 +591,7 @@
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g3 -w -O2 -Olimit 1500"
 
 # Note: sia_become_user() used by LOGINPW=os4 doesn't seem to work right.  The
@@ -572,6 +601,7 @@
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CHECKPW=os4 LOGINPW=os4 CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g3 -w -std0 -O2"
 
 osx:	# Mac OS X
@@ -579,6 +609,7 @@
 	 CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool MAILSPOOL=/var/mail \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCC4CFLAGS)"
 
 ptx:	# PTX
@@ -587,12 +618,13 @@
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/resh \
 	 BASECFLAGS="-Wc,-O3 -Wc,-seq -Dprivate=PRIVATE" \
-	 BASELDFLAGS="-lseq -lsec -lsocket -linet -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lseq -lsec -lsocket -linet -lnsl -lgen" \
 	 RANLIB=true
 
 pyr:	# Pyramid
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -Dconst="
 
 qnx:	# QNX
@@ -600,12 +632,13 @@
 	 CHECKPW=psx LOGINPW=old \
 	 RSHPATH=/usr/ucb/rsh \
 	 BASECFLAGS="-Otax -g -Dunix=1 -D_POSIX_SOURCE" \
-	 BASELDFLAGS="-g -N128k -llogin -lsocket -lunix"
+	 BASELDFLAGS="../c-client/client.a -g -N128k -llogin -lsocket -lunix"
 
 s40:	# SUN-OS 4.0
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
-	 BASECFLAGS="-g -Dconst="
+	 BASECFLAGS="-g -Dconst=" \
+	 BASELDFLAGS="../c-client.a" 
 
 sc5:	# SCO Open Server 5.0
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -615,7 +648,7 @@
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-O3 -s -belf" \
-	 BASELDFLAGS="-lsocket -lprot -lx -ltinfo -lm" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lx -ltinfo -lm" \
 	 RANLIB=true
 
 sco:	# Santa Cruz Operation
@@ -624,7 +657,7 @@
 	 CREATEPROTO=mmdfproto EMPTYPROTO=mmdfproto \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-O3" \
-	 BASELDFLAGS="-lsocket -lprot -lcrypt_i -lx -los" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lcrypt_i -lx -los" \
 	 RANLIB=true LN=ln
 
 # Note: setting _POSIX_SOURCE doesn't seem to build it as of SGI IRIX 5.3
@@ -635,6 +668,7 @@
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bsd/rsh \
 	 BASECFLAGS="-g3 -O2 -Olimit 8192" \
+	 BASELDFALGS="../c-client/c-client.a" \
 	 RANLIB=true
 
 sg6:	# Silicon Graphics, IRIX 6.5
@@ -643,6 +677,7 @@
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bsd/rsh \
 	 BASECFLAGS="-g3 -O2 -OPT:Olimit=0 -woff 1110,1116" \
+	 BASELDFALGS="../c-client/c-client.a" \
 	 RANLIB=true
 
 # Note: Mark Kaesling says that setluid() isn't in HP-UX with SecureWare.
@@ -654,7 +689,7 @@
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Ae" \
-	 BASELDFLAGS="-lnet -lV3 -lsec" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3 -lsec" \
 	 RANLIB=true
 
 slx:	# Secure Linux
@@ -668,7 +703,7 @@
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="$(LDOPT) -lpam -ldl -lcrypt"
+	 BASELDFLAGS="../c-client/c-client.a $(LDOPT) -lpam -ldl -lcrypt"
 
 sl4:	# Secure Linux using libc4
 	@echo You are building for libc4 versions of Secure Linux
@@ -680,7 +715,7 @@
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lshadow"
+	 BASELDFLAGS="../c-client/c-client.a -lshadow"
 
 sl5:	# Secure Linux using libc5
 	@echo You are building for libc5 versions of Secure Linux
@@ -691,6 +726,7 @@
 	 SPOOLDIR=/var/spool \
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 snx:	# Siemens Nixdorf SINIX and Reliant UNIX
@@ -700,7 +736,7 @@
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -D_SYS_CLOCK_H -Dconst=" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true
 
 # Sorry about the -w, but the cretinous SUN Workshop Pro C compiler barfs on
@@ -761,20 +797,20 @@
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -O2" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true CC=ucbcc
 
 sos:	# Secure OSF/1
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CHECKPW=sce LOGINPW=sec CRXTYPE=nfs \
 	 BASECFLAGS="-g3 -w -O2 -Olimit 1500" \
-	 BASELDFLAGS="-lsecurity -laud"
+	 BASELDFLAGS="../c-client/c-client.a -lsecurity -laud"
 
 ssn:	# Secure SUN-OS
 	$(BUILD) `$(CAT) SPECIALS` OS=sun \
 	 CHECKPW=ssn CRXTYPE=nfs \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-ldl"
+	 BASELDFLAGS="../c-client/c-client.a -ldl"
 
 sua:	# Windows Vista Subsystem for UNIX Applications
 	$(BUILD) `$(CAT) SPECIALS` OS=sua \
@@ -783,13 +819,13 @@
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -O2" \
-	 BASELDFLAGS="-lcrypt"
+	 BASELDFLAGS="../c-client/c-client.a -lcrypt"
 
 sun:	# SUN-OS
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-ldl"
+	 BASELDFLAGS="../c-client/c-client.a -ldl"
 
 sv2:	# SVR2
 	@echo You are being *very* optimistic!
@@ -798,7 +834,7 @@
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-Dconst= -DSYSTEM5 -DSHORT_IDENT -I/usr/ethernet/include" \
-	 BASELDFLAGS="-lnet" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet" \
 	 RANLIB=true LN=ln
 
 sv4:	# SVR4
@@ -808,14 +844,14 @@
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/resh \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true
 
 ult:	# Ultrix
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CHECKPW=ult CRXTYPE=nfs \
 	 BASECFLAGS="-g3 -O2 -Olimit 1500 -Dconst=" \
-	 BASELDFLAGS="-lauth -lc"
+	 BASELDFLAGS="../c-client/c-client.a -lauth -lc"
 
 uw2:	# UnixWare SVR4.2
 	$(BUILD) `$(CAT) SPECIALS` OS=sv4 \
@@ -824,18 +860,19 @@
 	 ACTIVEFILE=/var/news/lib/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true
 
 vul:	# VAX Ultrix
 	$(BUILD) `$(CAT) SPECIALS` OS=ult \
 	 SIGTYPE=psx CHECKPW=ult CRXTYPE=nfs \
 	 BASECFLAGS="-O2 -Dconst=" \
-	 BASELDFLAGS="-lauth -lc"
+	 BASELDFLAGS="../c-client/c-client.a -lauth -lc"
 
 vu2:	# VAX Ultrix 2.3, etc.
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-O2 -Dconst= -Dvoid=char"
 
 
@@ -857,7 +894,7 @@
 # Cleanup
 
 clean:
-	sh -c '$(RM) auths.c crexcl.c ip_unix.c linkage.[ch] siglocal.c osdep*.[ch] *.o ARCHIVE *FLAGS *TYPE $(ARCHIVE) || true'
+	sh -c '$(RM) auths.c crexcl.c ip_unix.c linkage.[ch] siglocal.c osdep*.[ch] *.o ARCHIVE ARCHIVENAME *FLAGS *TYPE $(ARCHIVE) || true'
 
 
 # Dependencies
@@ -964,6 +1001,7 @@
 	echo $(BASELDFLAGS) $(EXTRALDFLAGS) > LDFLAGS
 	echo "$(ARRC) $(ARCHIVE) $(BINARIES);$(RANLIB) $(ARCHIVE)" > ARCHIVE
 	echo $(OS) > OSTYPE
+	echo $(ARCHIVE) >ARCHIVENAME
 	./drivers $(EXTRADRIVERS) $(DEFAULTDRIVERS) dummy
 	./mkauths $(EXTRAAUTHENTICATORS) $(DEFAULTAUTHENTICATORS)
 	echo "  mail_versioncheck (CCLIENTVERSION);" >> linkage.c
diff -ur imap-2004b.orig/src/tmail/Makefile imap-2004b/src/tmail/Makefile
--- imap-2004b.orig/src/tmail/Makefile	2002-11-19 01:45:14.000000000 +0100
+++ imap-2004b/src/tmail/Makefile	2005-01-11 00:18:13.000000000 +0100
@@ -18,14 +18,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 tmail: $(CCLIENTLIB) tmail.o
 	$(CC) $(CFLAGS) -o tmail tmail.o $(LDFLAGS)
