diff -Naur imap-2001.BETA.SNAP-0107022325/src/imapd/imapd.c imap-2001.BETA.SNAP-0107022325-p/src/imapd/imapd.c
--- imap-2001.BETA.SNAP-0107022325/src/imapd/imapd.c	Fri Jun 22 21:48:53 2001
+++ imap-2001.BETA.SNAP-0107022325-p/src/imapd/imapd.c	Tue Jan  8 13:41:17 2002
@@ -468,7 +468,8 @@
 				/* start TLS security */
 	else if (!strcmp (cmd,"STARTTLS")) {
 	  if (arg) response = badarg;
-	  else if (lsterr = ssl_start_tls (argv[0])) response = lose;
+	  else if (lsterr = ssl_start_tls (((s = strrchr (argv[0],'/')) || (s = strrchr (argv[0],'\\'))) ?
+	      s+1 : argv[0])) response = lose;
 	}
 	else response =
 	  "%.80s BAD Command unrecognized/login please: %.80s\015\012";
