--- imap-4.7c/src/ipopd/ipop2d.c~	Wed Nov 17 03:31:25 1999
+++ imap-4.7c/src/ipopd/ipop2d.c	Thu Aug 31 18:58:57 2000
@@ -302,7 +302,10 @@
 #endif
   if (!s && server_login (user = cpystr (u),pass,argc,argv)) {
     syslog (LOG_INFO,"Login user=%.80s host=%.80s",user,tcp_clienthost ());
-    strcpy (tmp,"INBOX");	/* local; attempt login, select INBOX */
+    if (argc > 1)
+	    strcpy(tmp,argv[1]);
+    else
+    	strcpy (tmp,"INBOX");	/* local; attempt login, select INBOX */
   }
   else {
     fputs ("- Bad login\015\012",stdout);
