diff -ur imap-2004b.orig/src/osdep/unix/os_lnx.h imap-2004b/src/osdep/unix/os_lnx.h
--- imap-2004b.orig/src/osdep/unix/os_lnx.h	2005-01-11 00:16:37.000000000 +0100
+++ imap-2004b/src/osdep/unix/os_lnx.h	2005-01-11 00:41:52.000000000 +0100
@@ -18,19 +18,6 @@
  * CPYRIGHT, included with this Distribution.
  */
 
-/*
- *** These lines are claimed to be necessary to build on Debian Linux on an
- *** Alpha.
- */
-
-#ifndef _XOPEN_SOURCE
-#define _XOPEN_SOURCE 1
-#endif /* _XOPEN_SOURCE */
-#ifndef _BSD_SOURCE
-#define _BSD_SOURCE 1
-#endif /* _BSD_SOURCE */
-
-/* end Debian Linux on Alpha strangeness */
 
 #include <stdlib.h>
 #include <string.h>
diff -ur imap-2004b.orig/src/osdep/unix/os_slx.h imap-2004b/src/osdep/unix/os_slx.h
--- imap-2004b.orig/src/osdep/unix/os_slx.h	2005-01-11 00:16:37.000000000 +0100
+++ imap-2004b/src/osdep/unix/os_slx.h	2005-01-11 00:42:11.000000000 +0100
@@ -18,19 +18,6 @@
  * CPYRIGHT, included with this Distribution.
  */
 
-/*
- *** These lines are claimed to be necessary to build on Debian Linux on an
- *** Alpha.
- */
-
-#ifndef _XOPEN_SOURCE
-#define _XOPEN_SOURCE 1
-#endif /* _XOPEN_SOURCE */
-#ifndef _BSD_SOURCE
-#define _BSD_SOURCE 1
-#endif /* _BSD_SOURCE */
-
-/* end Debian Linux on Alpha strangeness */
 
 #include <stdlib.h>
 #include <string.h>
