--- imap-2001.BETA.SNAP-0106191041.orig/src/ipopd/ipop2d.c	Sat Apr 14 00:37:23 2001
+++ imap-2001.BETA.SNAP-0106191041/src/ipopd/ipop2d.c	Mon Jun 25 15:16:33 2001
@@ -277,7 +277,10 @@
     if (server_login (user = cpystr (u),pass,s,argc,argv)) {
       syslog (LOG_INFO,"%sLogin user=%.80s host=%.80s",s ? "Admin " : "",
 	      user,tcp_clienthost ());
-      return c_fold ("INBOX");	/* local; select INBOX */
+      if (argc > 1)
+	return argv[1];
+      else
+      	return c_fold ("INBOX");	/* local; select INBOX */
     }
   }
 #ifndef DISABLE_POP_PROXY
