diff -ur imap-2004/src/imapd/imapd.c imap-patched/src/imapd/imapd.c
--- imap-2004/src/imapd/imapd.c	Sun May 30 07:14:15 2004
+++ imap-patched/src/imapd/imapd.c	Sun May 30 07:16:59 2004
@@ -183,7 +183,7 @@
 
 /* Global storage */
 
-char *version = "2004.350";	/* version number of this server */
+char *version = "2004.350PLD";	/* version number of this server */
 time_t alerttime = 0;		/* time of last alert */
 time_t sysalerttime = 0;	/* time of last system alert */
 time_t useralerttime = 0;	/* time of last user alert */
diff -ur imap-2004/src/ipopd/ipop2d.c imap-patched/src/ipopd/ipop2d.c
--- imap-2004/src/ipopd/ipop2d.c	Sun May 30 06:35:12 2004
+++ imap-patched/src/ipopd/ipop2d.c	Sun May 30 07:16:29 2004
@@ -51,7 +51,7 @@
 
 /* Global storage */
 
-char *version = "2004.67";	/* server version */
+char *version = "2004.67PLD";	/* server version */
 short state = LISN;		/* server state */
 short critical = NIL;		/* non-zero if in critical code */
 MAILSTREAM *stream = NIL;	/* mailbox stream */
diff -ur imap-2004/src/ipopd/ipop3d.c imap-patched/src/ipopd/ipop3d.c
--- imap-2004/src/ipopd/ipop3d.c	Tue Apr 27 21:50:29 2004
+++ imap-patched/src/ipopd/ipop3d.c	Sun May 30 07:16:42 2004
@@ -57,7 +57,7 @@
 
 /* Global storage */
 
-char *version = "2004.87";	/* server version */
+char *version = "2004.87PLD";	/* server version */
 short state = AUTHORIZATION;	/* server state */
 short critical = NIL;		/* non-zero if in critical code */
 MAILSTREAM *stream = NIL;	/* mailbox stream */
