diff -urNp -x '*.orig' imap-2007f.org/src/osdep/unix/Makefile imap-2007f/src/osdep/unix/Makefile
--- imap-2007f.org/src/osdep/unix/Makefile	2021-09-29 21:57:46.632676218 +0200
+++ imap-2007f/src/osdep/unix/Makefile	2021-09-29 21:57:46.699342738 +0200
@@ -506,7 +506,7 @@ isc:	# Interactive
 lnps:  # Linux Pluggable Authentication modules (c-client as shared lib)
 	$(BUILD) `cat EXTRASPECIALS` OS=lnx \
 	SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
-	SPOOLDIR=/var/spool \
+	SPOOLDIR=/var/spool MAILSPOOL=/var/mail \
 	ACTIVEFILE=/var/lib/news/active \
 	RSHPATH=/usr/bin/rsh \
 	BASECFLAGS="$(GCCCFLAGS)" \
@@ -518,7 +518,7 @@ lnps:  # Linux Pluggable Authentication
 lnp:	# Linux Pluggable Authentication modules
 	$(BUILD) `$(CAT) SPECIALS` OS=slx \
 	 SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
-	 SPOOLDIR=/var/spool \
+	 SPOOLDIR=/var/spool MAILSPOOL=/var/mail \
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
