diff -urNp -x '*.orig' imap-2007f.org/src/osdep/unix/os_lnx.h imap-2007f/src/osdep/unix/os_lnx.h
--- imap-2007f.org/src/osdep/unix/os_lnx.h	2021-09-29 21:57:47.276008132 +0200
+++ imap-2007f/src/osdep/unix/os_lnx.h	2021-09-29 21:57:47.349341303 +0200
@@ -26,19 +26,6 @@
  * Last Edited:	30 August 2006
  */
 
-/*
- *** These lines are claimed to be necessary to build on Debian Linux on an
- *** Alpha.
- */
-
-#ifndef _XOPEN_SOURCE
-#define _XOPEN_SOURCE 1
-#endif /* _XOPEN_SOURCE */
-#ifndef _BSD_SOURCE
-#define _BSD_SOURCE 1
-#endif /* _BSD_SOURCE */
-
-/* end Debian Linux on Alpha strangeness */
 
 #include <stdlib.h>
 #include <string.h>
diff -urNp -x '*.orig' imap-2007f.org/src/osdep/unix/os_slx.h imap-2007f/src/osdep/unix/os_slx.h
--- imap-2007f.org/src/osdep/unix/os_slx.h	2021-09-29 21:57:47.276008132 +0200
+++ imap-2007f/src/osdep/unix/os_slx.h	2021-09-29 21:57:47.349341303 +0200
@@ -26,19 +26,6 @@
  * Last Edited:	30 August 2006
  */
 
-/*
- *** These lines are claimed to be necessary to build on Debian Linux on an
- *** Alpha.
- */
-
-#ifndef _XOPEN_SOURCE
-#define _XOPEN_SOURCE 1
-#endif /* _XOPEN_SOURCE */
-#ifndef _BSD_SOURCE
-#define _BSD_SOURCE 1
-#endif /* _BSD_SOURCE */
-
-/* end Debian Linux on Alpha strangeness */
 
 #include <stdlib.h>
 #include <string.h>
