--- imap-2006c1/src/imapd/imapd.c.orig	2006-10-17 22:27:51.000000000 +0200
+++ imap-2006c1/src/imapd/imapd.c	2006-11-21 11:44:25.019505000 +0100
@@ -202,7 +202,7 @@
 
 /* Global storage */
 
-char *version = "2006d.376";	/* version number of this server */
+char *version = "2006d.376PLD";	/* version number of this server */
 char *logout = "Logout";	/* syslogreason for logout */
 char *goodbye = NIL;		/* bye reason */
 time_t alerttime = 0;		/* time of last alert */
--- imap-2006c1/src/ipopd/ipop2d.c.orig	2006-11-21 11:42:50.109505000 +0100
+++ imap-2006c1/src/ipopd/ipop2d.c	2006-11-21 11:44:43.599505000 +0100
@@ -59,7 +59,7 @@
 
 /* Global storage */
 
-char *version = "2006d.72";	/* server version */
+char *version = "2006d.72PLD";	/* server version */
 short state = LISN;		/* server state */
 short critical = NIL;		/* non-zero if in critical code */
 MAILSTREAM *stream = NIL;	/* mailbox stream */
--- imap-2006c1/src/ipopd/ipop3d.c.orig	2006-10-12 19:30:55.000000000 +0200
+++ imap-2006c1/src/ipopd/ipop3d.c	2006-11-21 11:44:55.579505000 +0100
@@ -65,7 +65,7 @@
 
 /* Global storage */
 
-char *version = "2006d.96";	/* server version */
+char *version = "2006d.96PLD";	/* server version */
 short state = AUTHORIZATION;	/* server state */
 short critical = NIL;		/* non-zero if in critical code */
 MAILSTREAM *stream = NIL;	/* mailbox stream */
