diff -ur imap-2004/src/osdep/unix/Makefile imap-patched/src/osdep/unix/Makefile
--- imap-2004/src/osdep/unix/Makefile	Fri Apr 30 02:08:48 2004
+++ imap-patched/src/osdep/unix/Makefile	Sun May 30 06:33:23 2004
@@ -29,11 +29,11 @@
 
 # Extended flags needed for SSL.  You may need to modify.
 
-SSLDIR=/usr/local/ssl
-SSLCERTS=$(SSLDIR)/certs
-SSLKEYS=$(SSLCERTS)
-SSLINCLUDE=$(SSLDIR)/include
-SSLLIB=$(SSLDIR)/lib
+SSLDIR=/usr/lib/openssl
+SSLCERTS=/var/lib/openssl/certs
+SSLKEYS=/var/lib/openssl/certs
+SSLINCLUDE=/usr/include
+SSLLIB=/usr/lib
 
 SSLCRYPTO=-lcrypto
 
@@ -620,12 +620,13 @@
 	@echo If you want libc5 versions you must use sl5 instead!
 	@echo If you want libc4 versions you must use sl4 instead!
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
-	 SIGTYPE=psx CHECKPW=psx CRXTYPE=nfs \
+	 SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool \
+	 MAILSPOOL=/var/mail \
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
-	 BASECFLAGS="-g -fno-omit-frame-pointer $(GCCOPTLEVEL)" \
-	 BASELDFLAGS="-lcrypt"
+	 BASECFLAGS="$(OPT)" \
+	 BASELDFLAGS="$(LDOPT) -lpam -ldl -lcrypt"
 
 sl4:	# Secure Linux using libc4
 	@echo You are building for libc4 versions of Secure Linux
diff -ur imap-2004/src/osdep/unix/os_lnx.h imap-patched/src/osdep/unix/os_lnx.h
--- imap-2004/src/osdep/unix/os_lnx.h	Tue Jul 15 22:53:32 2003
+++ imap-patched/src/osdep/unix/os_lnx.h	Sun May 30 06:31:26 2004
@@ -37,7 +37,7 @@
 #include <fcntl.h>
 #include <syslog.h>
 #include <sys/file.h>
-
+#include <ustat.h>
 
 /* Linux gets this wrong */
 
diff -ur imap-2004/src/osdep/unix/os_slx.h imap-patched/src/osdep/unix/os_slx.h
--- imap-2004/src/osdep/unix/os_slx.h	Tue Jul 15 22:53:32 2003
+++ imap-patched/src/osdep/unix/os_slx.h	Sun May 30 06:31:26 2004
@@ -37,7 +37,7 @@
 #include <fcntl.h>
 #include <syslog.h>
 #include <sys/file.h>
-
+#include <ustat.h>
 
 /* Linux gets this wrong */
 
