diff -ur imap-2001a/src/osdep/unix/env_unix.c imap-2001a.new/src/osdep/unix/env_unix.c
--- imap-2001a/src/osdep/unix/env_unix.c	Thu Oct 18 05:27:59 2001
+++ imap-2001a.new/src/osdep/unix/env_unix.c	Fri Jun 14 00:38:18 2002
@@ -1001,12 +1001,6 @@
       }
       close (pi[0]); close (pi[1]);
     }
-    if (lockEaccesError) {	/* punt silently if paranoid site */
-      sprintf (tmp,"Mailbox vulnerable - directory %.80s",base->lock);
-      if (s = strrchr (tmp,'/')) *s = '\0';
-      strcat (tmp," must have 1777 protection");
-      MM_LOG (tmp,WARN);
-    }
     break;
   default:
     sprintf (tmp,"Mailbox vulnerable - error creating %.80s: %s",
Only in imap-2001a.new/src/osdep/unix: env_unix.c~
