--- imap-2006c1/src/osdep/unix/Makefile.orig	2006-10-23 23:35:38.000000000 +0200
+++ imap-2006c1/src/osdep/unix/Makefile	2006-11-21 11:14:20.849505000 +0100
@@ -48,17 +48,17 @@
 # Try to have some consistency in GCC builds.  We want optimization, but we
 # also want to be able to debug.
 
-GCCCFLAGS= -g $(GCCOPTLEVEL) -pipe -fno-omit-frame-pointer
+GCCCFLAGS= $(GCCOPTLEVEL) -pipe
 GCC4CFLAGS= $(GCCCFLAGS) -Wno-pointer-sign
 
 
 # Extended flags needed for SSL.  You may need to modify.
 
-SSLDIR=/usr/local/ssl
-SSLCERTS=$(SSLDIR)/certs
-SSLKEYS=$(SSLCERTS)
-SSLINCLUDE=$(SSLDIR)/include
-SSLLIB=$(SSLDIR)/lib
+SSLDIR=/usr/lib/openssl
+SSLCERTS=/var/lib/openssl/certs
+SSLKEYS=/var/lib/openssl/certs
+SSLINCLUDE=/usr/include
+SSLLIB=/usr/lib
 
 SSLCRYPTO=-lcrypto
 
@@ -656,12 +656,13 @@
 	@echo If you want libc5 versions you must use sl5 instead!
 	@echo If you want libc4 versions you must use sl4 instead!
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
-	 SIGTYPE=psx CHECKPW=psx CRXTYPE=nfs \
+	 SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool \
+	 MAILSPOOL=/var/mail \
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lcrypt"
+	 BASELDFLAGS="$(LDOPT) -lpam -ldl -lcrypt"
 
 sl4:	# Secure Linux using libc4
 	@echo You are building for libc4 versions of Secure Linux

