diff -urNp -x '*.orig' imap-2007f.org/Makefile imap-2007f/Makefile
--- imap-2007f.org/Makefile	2011-07-23 02:20:21.000000000 +0200
+++ imap-2007f/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -84,6 +84,7 @@
 # lnx	Linux with traditional passwords and crypt() in the C library
 #	 (see lnp, sl4, sl5, and slx)
 # lnp	Linux with Pluggable Authentication Modules (PAM)
+# lnps	Linux with Pluggable Authentication Modules (PAM) (with shared libc-client)
 # lmd	Mandrake Linux
 # lr5	RedHat Enterprise 5 and later (same as lfd)
 # lrh	RedHat Linux 7.2 and later
@@ -307,7 +308,7 @@ SPECIALS:
 
 # Note on SCO you may have to set LN to "ln".
 
-a32 a41 a52 aix bs3 bsi d-g d54 do4 drs epx ga4 gas gh9 ghp ghs go5 gsc gsg gso gul h11 hpp hpx lnp lyn mct mnt nec nto nxt nx3 osf os4 ptx qnx sc5 sco sgi sg6 shp sl4 sl5 slx snx soc sol sos uw2: an
+a32 a41 a52 aix bs3 bsi d-g d54 do4 drs epx ga4 gas gh9 ghp ghs go5 gsc gsg gso gul h11 hpp hpx lnp lnps lyn mct mnt nec nto nxt nx3 osf os4 ptx qnx sc5 sco sgi sg6 shp sl4 sl5 slx snx soc sol sos uw2: an
 	$(BUILD) BUILDTYPE=$@
 
 # If you use sv4, you may find that it works to move it to use the an process.
diff -urNp -x '*.orig' imap-2007f.org/src/dmail/Makefile imap-2007f/src/dmail/Makefile
--- imap-2007f.org/src/dmail/Makefile	2011-07-23 02:19:57.000000000 +0200
+++ imap-2007f/src/dmail/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -26,14 +26,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 dmail: $(CCLIENTLIB) dmail.o dquota.o
 	$(CC) $(CFLAGS) -o dmail dmail.o dquota.o $(LDFLAGS)
diff -urNp -x '*.orig' imap-2007f.org/src/imapd/Makefile imap-2007f/src/imapd/Makefile
--- imap-2007f.org/src/imapd/Makefile	2011-07-23 02:20:00.000000000 +0200
+++ imap-2007f/src/imapd/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -43,12 +43,12 @@ SHELL= /bin/sh
 # Get local definitions from c-client directory
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS` $(NSBD) $(ENBD) -DANOFILE=\"$(ANO)\" \
 	-DALERTFILE=\"$(ALERT)\" -DNNTPFILE=\"$(NNTP)\" \
 	-DUSERALERTFILE=\"$(USERALERT)\" -DSHUTDOWNFILE=\"$(SHUTDOWN)\"
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 all:	imapd
 
diff -urNp -x '*.orig' imap-2007f.org/src/ipopd/Makefile imap-2007f/src/ipopd/Makefile
--- imap-2007f.org/src/ipopd/Makefile	2011-07-23 02:20:17.000000000 +0200
+++ imap-2007f/src/ipopd/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -26,14 +26,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 ipopd: ipop2d ipop3d
 
diff -urNp -x '*.orig' imap-2007f.org/src/mailutil/Makefile imap-2007f/src/mailutil/Makefile
--- imap-2007f.org/src/mailutil/Makefile	2011-07-23 02:19:56.000000000 +0200
+++ imap-2007f/src/mailutil/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -26,14 +26,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 mailutil: $(CCLIENTLIB) mailutil.o
 	$(CC) $(CFLAGS) -o mailutil mailutil.o $(LDFLAGS)
diff -urNp -x '*.orig' imap-2007f.org/src/mtest/Makefile imap-2007f/src/mtest/Makefile
--- imap-2007f.org/src/mtest/Makefile	2011-07-23 02:20:17.000000000 +0200
+++ imap-2007f/src/mtest/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -26,14 +26,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 all:	mtest
 
diff -urNp -x '*.orig' imap-2007f.org/src/osdep/unix/Makefile imap-2007f/src/osdep/unix/Makefile
--- imap-2007f.org/src/osdep/unix/Makefile	2021-09-29 21:57:46.299343621 +0200
+++ imap-2007f/src/osdep/unix/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -189,7 +189,7 @@ a32:	# AIX 3.2 for RS/6000
 	 SPOOLDIR=/var/spool \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -Dunix=1 -D_BSD" \
-	 BASELDFLAGS="-lbsd"
+	 BASELDFLAGS="../c-client/c-client.a -lbsd"
 
 a41:	# AIX 4.1 for RS/6000
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -213,12 +213,13 @@ aix:	# AIX/370
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="-g" \
-	 BASELDFLAGS="-lbsd"
+	 BASELDFLAGS="../c-client/c-client.a -lbsd"
 
 aos:	# AOS for RT
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
-	 BASECFLAGS="-g -Dconst="
+	 BASECFLAGS="-g -Dconst=" \
+	 BASELDFLAGS="../c-client/c-client.a"
 
 art:	# AIX 2.2.1 for RT
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -227,7 +228,8 @@ art:	# AIX 2.2.1 for RT
 	 ACTIVEFILE=/usr/local/news/control/active \
 	 RSHPATH=/bin/rsh \
 	 BASECFLAGS="-g -Dconst= -Dvoid=char" \
-	 RANLIB=true 
+	 RANLIB=true \
+	 BASELDFLAGS="../c-client/c-client.a"
 
 asv:	# Altos SVR4
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -235,7 +237,7 @@ asv:	# Altos SVR4
 	 ACTIVEFILE=/usr/spool/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-Dconst= -DSIGSTOP=SIGKILL" \
-	 BASELDFLAGS="-lsocket -lrpc -lgen -lcrypt -lxenix" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lrpc -lgen -lcrypt -lxenix" \
 	 RANLIB=true
 
 aux:	# A/UX
@@ -243,6 +245,7 @@ aux:	# A/UX
 	 CRXTYPE=nfs \
 	 MAILSPOOL=/usr/mail \
 	 BASECFLAGS="-g -B/usr/lib/big/ -Dvoid=char -Dconst=" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true ARRC="ar -rc"
 
 bs3:	# BSD/i386 3.0 or higher
@@ -251,11 +254,13 @@ bs3:	# BSD/i386 3.0 or higher
 	 SPOOLDIR=/var NEWSSPOOL=/var/news/spool \
 	 ACTIVEFILE=/var/news/etc/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)" CC=shlicc
 
 bsd:	# BSD UNIX
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -Dconst="
 
 bsf:	# FreeBSD
@@ -273,6 +278,7 @@ bsi:	# BSD/i386
 	 SPOOLDIR=/var NEWSSPOOL=/var/news/spool \
 	 ACTIVEFILE=/var/news/etc/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 bso:	# OpenBSD
@@ -281,6 +287,7 @@ bso:	# OpenBSD
 	 SPOOLDIR=/var \
 	 ACTIVEFILE=/usr/local/news/lib/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 cvx:	# Convex
@@ -306,7 +313,7 @@ d-g:	# Data General DG/UX
 	 ACTIVEFILE=/local/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-lnsl -lsocket" \
+	 BASELDFLAGS="../c-client/c-client.a -lnsl -lsocket" \
 	 RANLIB=true
 
 d54:	# Data General DG/UX 5.4
@@ -316,7 +323,7 @@ d54:	# Data General DG/UX 5.4
 	 ACTIVEFILE=/local/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-lnsl -lsocket" \
+	 BASELDFLAGS="../c-client/c-client.a -lnsl -lsocket" \
 	 RANLIB=true
 
 dpx:	# Bull DPX/2
@@ -324,7 +331,7 @@ dpx:	# Bull DPX/2
 	 SIGTYPE=sv4 CHECKPW=sv4 LOGINPW=sv4 \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-Dconst= -DSYSTEM5 -DSHORT_IDENT" \
-	 BASELDFLAGS="-linet" \
+	 BASELDFLAGS="../c-client/c-client.a -linet" \
 	 RANLIB=true LN=ln
 
 drs:	# ICL DRS/NX
@@ -334,18 +341,20 @@ drs:	# ICL DRS/NX
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-O" \
-	 BASELDFLAGS="-lsocket -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lgen" \
 	 RANLIB=true
 
 do4:	# Apollo Domain/OS sr10.4
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="-A systype,bsd4.3 -D_APOLLO_SOURCE" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true
 
 dyn:	# Dynix
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -Dconst="
 
 epx:	# EP/IX
@@ -364,7 +373,7 @@ ga4:	# GCC AIX 4.1 for RS/6000
 	 SPOOLDIR=/var/spool \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -Dunix=1 -D_BSD" \
-	 BASELDFLAGS="-ls"
+	 BASELDFLAGS="../c-client/c-client.a -ls"
 
 gas:	# GCC Altos SVR4
 	$(BUILD) `$(CAT) SPECIALS` OS=asv \
@@ -372,7 +381,7 @@ gas:	# GCC Altos SVR4
 	 ACTIVEFILE=/usr/spool/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-g -O -DALTOS_SYSTEM_V -DSIGSTOP=SIGKILL" \
-	 BASELDFLAGS="-lsocket -lrpc -lgen -lcrypt -lxenix" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lrpc -lgen -lcrypt -lxenix" \
 	 RANLIB=true CC=gcc
 
 gh9:	# GCC HP-UX 9.x
@@ -381,6 +390,7 @@ gh9:	# GCC HP-UX 9.x
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true CC=gcc
 
 ghp:	# GCC HP-UX 10.x
@@ -390,6 +400,7 @@ ghp:	# GCC HP-UX 10.x
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true CC=gcc
 
 ghs:	# GCC HP-UX with Trusted Computer Base
@@ -399,7 +410,7 @@ ghs:	# GCC HP-UX with Trusted Computer B
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lnet -lV3 -lsec" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3 -lsec" \
 	 RANLIB=true CC=gcc
 
 go5:	# GCC 2.7.1 (95q4) SCO Open Server 5.0.x
@@ -410,7 +421,7 @@ go5:	# GCC 2.7.1 (95q4) SCO Open Server
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="$(GCCCFLAGS) -I/usr/include -L/lib" \
-	 BASELDFLAGS="-lsocket -lprot -lx -ltinfo -lm" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lx -ltinfo -lm" \
 	 RANLIB=true CC=gcc
 
 gsc:	# Santa Cruz Operation
@@ -419,7 +430,7 @@ gsc:	# Santa Cruz Operation
 	 CREATEPROTO=mmdfproto EMPTYPROTO=mmdfproto \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lsocket -lprot -lcrypt_i -lx -los" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lcrypt_i -lx -los" \
 	 RANLIB=true LN=ln CC=gcc
 
 gsg:	# GCC Silicon Graphics
@@ -428,6 +439,7 @@ gsg:	# GCC Silicon Graphics
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bsd/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 RANLIB=true CC=gcc
 
 gso:	os_sol.h	# GCC Solaris
@@ -437,21 +449,21 @@ gso:	os_sol.h	# GCC Solaris
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true CC=gcc
 
 gsu:	# GCC SUN-OS
 	$(BUILD) `$(CAT) SPECIALS` OS=sun \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-ldl" \
+	 BASELDFLAGS="../c-client/c-client.a -ldl" \
 	 CC=gcc
 
 gul:	# GCC Ultrix
 	$(BUILD) `$(CAT) SPECIALS` OS=ult \
 	 SIGTYPE=psx CHECKPW=ult CRXTYPE=nfs \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lauth -lc" \
+	 BASELDFLAGS="../c-client/c-client.a -lauth -lc" \
 	 CC=gcc
 
 h11:	# HP-UX 11i
@@ -469,7 +481,7 @@ hpp:	# HP-UX 9.x
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Aa -D_HPUX_SOURCE" \
-	 BASELDFLAGS="-lnet -lV3" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3" \
 	 RANLIB=true
 
 hpx:	# HP-UX 10.x
@@ -479,7 +491,7 @@ hpx:	# HP-UX 10.x
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Ae" \
-	 BASELDFLAGS="-lnet -lV3" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3" \
 	 RANLIB=true
 
 isc:	# Interactive
@@ -488,9 +500,21 @@ isc:	# Interactive
 	 SPOOLDIR=/var/spool MAILSPOOL=/var/mail \
 	 ACTIVEFILE=/var/spool/news/active \
 	 BASECFLAGS="-Xp -D_SYSV3" \
-	 BASELDFLAGS="-linet -lnsl_s -lgen -lx -lsec -liberty" \
+	 BASELDFLAGS="../c-client/c-client.a -linet -lnsl_s -lgen -lx -lsec -liberty" \
 	 RANLIB=true
 
+lnps:  # Linux Pluggable Authentication modules (c-client as shared lib)
+	$(BUILD) `cat EXTRASPECIALS` OS=lnx \
+	SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
+	SPOOLDIR=/var/spool \
+	ACTIVEFILE=/var/lib/news/active \
+	RSHPATH=/usr/bin/rsh \
+	BASECFLAGS="$(GCCCFLAGS)" \
+	ARCHIVE="libc-client.so" \
+	BASELDFLAGS="$(LDOPT) -L../c-client -lc-client -lpam -ldl -lc" \
+	ARRC="gcc --shared -Wl,-soname,libc-client.so.$(VERSION) -lssl -lcrypto -lpam -ldl -lc -o" \
+	RANLIB=true
+
 lnp:	# Linux Pluggable Authentication modules
 	$(BUILD) `$(CAT) SPECIALS` OS=slx \
 	 SIGTYPE=psx CHECKPW=pam CRXTYPE=nfs \
@@ -498,7 +522,7 @@ lnp:	# Linux Pluggable Authentication mo
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="$(PAMLDFLAGS)"
+	 BASELDFLAGS="../c-client/c-client.a $(PAMLDFLAGS)"
 
 lnx:	# Linux non-shadow passwords
 	@echo You are building for traditional Linux *without* shadow
@@ -517,12 +541,13 @@ lyn:	# LynxOS
 	 CRXTYPE=nfs \
 	 RSHPATH=/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS=-lbsd \
+	 BASELDFLAGS="../c-client/c-client.a -lbsd" \
 	 CC=gcc
 
 mct:	# MachTen - CRXTYPE=nfs doesn't work (at least not on 2.2)
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SPOOLDIR=/var/spool \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 mnt:	# Mint
@@ -540,7 +565,7 @@ neb:	# NetBSD
 	 ACTIVEFILE=/var/db/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lcrypt"
+	 BASELDFLAGS="../c-client/c-client.a -lcrypt"
 
 nec:	# NEC UX
 	$(BUILD) `$(CAT) SPECIALS` OS=sv4 \
@@ -549,7 +574,7 @@ nec:	# NEC UX
 	 ACTIVEFILE=/var/news/lib/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -Kopt=2 -KOlimit=2000" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true CC=/usr/abiccs/bin/cc
 
 nto:	# QNX Neutrino RTP
@@ -558,16 +583,19 @@ nto:	# QNX Neutrino RTP
 	 SPOOLDIR=/var/spool \
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -O"
 
 nxt:	# NEXTSTEP
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 nx3:	# NEXTSTEP 3.0 single threaded
 	$(BUILD) `$(CAT) SPECIALS` OS=nxt \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 	echo "void malloc_singlethreaded (void);" >> linkage.h
 	echo "  malloc_singlethreaded ();" >> linkage.c
@@ -576,6 +604,7 @@ osf:	# OSF/1
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g3 -w -O2 -Olimit 1500"
 
 # Note: sia_become_user() used by LOGINPW=os4 doesn't seem to work right.  The
@@ -585,6 +614,7 @@ os4:	# OSF/1 (Digital UNIX) 4
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CHECKPW=os4 LOGINPW=os4 CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g3 -w -std0 -O2"
 
 osx:	# Mac OS X
@@ -592,6 +622,7 @@ osx:	# Mac OS X
 	 CRXTYPE=nfs \
 	 SPOOLDIR=/var/spool MAILSPOOL=/var/mail \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCC4CFLAGS)"
 
 ptx:	# PTX
@@ -600,12 +631,13 @@ ptx:	# PTX
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/resh \
 	 BASECFLAGS="-Wc,-O3 -Wc,-seq -Dprivate=PRIVATE" \
-	 BASELDFLAGS="-lseq -lsec -lsocket -linet -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lseq -lsec -lsocket -linet -lnsl -lgen" \
 	 RANLIB=true
 
 pyr:	# Pyramid
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-g -Dconst="
 
 qnx:	# QNX
@@ -613,12 +645,13 @@ qnx:	# QNX
 	 CHECKPW=psx LOGINPW=old \
 	 RSHPATH=/usr/ucb/rsh \
 	 BASECFLAGS="-Otax -g -Dunix=1 -D_POSIX_SOURCE" \
-	 BASELDFLAGS="-g -N128k -llogin -lsocket -lunix"
+	 BASELDFLAGS="../c-client/client.a -g -N128k -llogin -lsocket -lunix"
 
 s40:	# SUN-OS 4.0
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
-	 BASECFLAGS="-g -Dconst="
+	 BASECFLAGS="-g -Dconst=" \
+	 BASELDFLAGS="../c-client.a" 
 
 sc5:	# SCO Open Server 5.0
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
@@ -628,7 +661,7 @@ sc5:	# SCO Open Server 5.0
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-O3 -s -belf" \
-	 BASELDFLAGS="-lsocket -lprot -lx -ltinfo -lm" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lx -ltinfo -lm" \
 	 RANLIB=true
 
 sco:	# Santa Cruz Operation
@@ -637,7 +670,7 @@ sco:	# Santa Cruz Operation
 	 CREATEPROTO=mmdfproto EMPTYPROTO=mmdfproto \
 	 RSHPATH=/usr/bin/rcmd \
 	 BASECFLAGS="-O3" \
-	 BASELDFLAGS="-lsocket -lprot -lcrypt_i -lx -los" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lprot -lcrypt_i -lx -los" \
 	 RANLIB=true LN=ln
 
 # Note: setting _POSIX_SOURCE doesn't seem to build it as of SGI IRIX 5.3
@@ -648,6 +681,7 @@ sgi:	# Silicon Graphics
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bsd/rsh \
 	 BASECFLAGS="-g3 -O2 -Olimit 8192" \
+	 BASELDFALGS="../c-client/c-client.a" \
 	 RANLIB=true
 
 sg6:	# Silicon Graphics, IRIX 6.5
@@ -656,6 +690,7 @@ sg6:	# Silicon Graphics, IRIX 6.5
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bsd/rsh \
 	 BASECFLAGS="-g3 -O2 -OPT:Olimit=0 -woff 1110,1116" \
+	 BASELDFALGS="../c-client/c-client.a" \
 	 RANLIB=true
 
 # Note: Mark Kaesling says that setluid() isn't in HP-UX with SecureWare.
@@ -667,7 +702,7 @@ shp:	# HP-UX with Trusted Computer Base
 	 ACTIVEFILE=/var/news/active \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-g -Ae" \
-	 BASELDFLAGS="-lnet -lV3 -lsec" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet -lV3 -lsec" \
 	 RANLIB=true
 
 slx:	# Secure Linux
@@ -681,7 +716,7 @@ slx:	# Secure Linux
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="$(LDOPT) -lpam -ldl -lcrypt"
+	 BASELDFLAGS="../c-client/c-client.a $(LDOPT) -lpam -ldl -lcrypt"
 
 sl4:	# Secure Linux using libc4
 	@echo You are building for libc4 versions of Secure Linux
@@ -693,7 +728,7 @@ sl4:	# Secure Linux using libc4
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="$(GCCCFLAGS)" \
-	 BASELDFLAGS="-lshadow"
+	 BASELDFLAGS="../c-client/c-client.a -lshadow"
 
 sl5:	# Secure Linux using libc5
 	@echo You are building for libc5 versions of Secure Linux
@@ -704,6 +739,7 @@ sl5:	# Secure Linux using libc5
 	 SPOOLDIR=/var/spool \
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="$(GCCCFLAGS)"
 
 snx:	# Siemens Nixdorf SINIX and Reliant UNIX
@@ -713,7 +749,7 @@ snx:	# Siemens Nixdorf SINIX and Reliant
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -D_SYS_CLOCK_H -Dconst=" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true
 
 # Sorry about the -w, but the cretinous SUN Workshop Pro C compiler barfs on
@@ -774,20 +810,20 @@ sol:	os_sol.h	# Solaris
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -O2" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true CC=ucbcc
 
 sos:	# Secure OSF/1
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CHECKPW=sce LOGINPW=sec CRXTYPE=nfs \
 	 BASECFLAGS="-g3 -w -O2 -Olimit 1500" \
-	 BASELDFLAGS="-lsecurity -laud"
+	 BASELDFLAGS="../c-client/c-client.a -lsecurity -laud"
 
 ssn:	# Secure SUN-OS
 	$(BUILD) `$(CAT) SPECIALS` OS=sun \
 	 CHECKPW=ssn CRXTYPE=nfs \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-ldl"
+	 BASELDFLAGS="../c-client/c-client.a -ldl"
 
 sua:	# Windows Vista Subsystem for UNIX Applications
 	$(BUILD) `$(CAT) SPECIALS` OS=sua \
@@ -796,13 +832,13 @@ sua:	# Windows Vista Subsystem for UNIX
 	 ACTIVEFILE=/var/lib/news/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g -O2" \
-	 BASELDFLAGS="-lcrypt"
+	 BASELDFLAGS="../c-client/c-client.a -lcrypt"
 
 sun:	# SUN-OS
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-ldl"
+	 BASELDFLAGS="../c-client/c-client.a -ldl"
 
 sv2:	# SVR2
 	@echo You are being *very* optimistic!
@@ -811,7 +847,7 @@ sv2:	# SVR2
 	 MAILSPOOL=/usr/mail \
 	 RSHPATH=/usr/bin/remsh \
 	 BASECFLAGS="-Dconst= -DSYSTEM5 -DSHORT_IDENT -I/usr/ethernet/include" \
-	 BASELDFLAGS="-lnet" \
+	 BASELDFLAGS="../c-client/c-client.a -lnet" \
 	 RANLIB=true LN=ln
 
 sv4:	# SVR4
@@ -821,14 +857,14 @@ sv4:	# SVR4
 	 ACTIVEFILE=/usr/share/news/active \
 	 RSHPATH=/usr/bin/resh \
 	 BASECFLAGS="-g -Dconst=" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true
 
 ult:	# Ultrix
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 SIGTYPE=psx CHECKPW=ult CRXTYPE=nfs \
 	 BASECFLAGS="-g3 -O2 -Olimit 1500 -Dconst=" \
-	 BASELDFLAGS="-lauth -lc"
+	 BASELDFLAGS="../c-client/c-client.a -lauth -lc"
 
 uw2:	# UnixWare SVR4.2
 	$(BUILD) `$(CAT) SPECIALS` OS=sv4 \
@@ -837,18 +873,19 @@ uw2:	# UnixWare SVR4.2
 	 ACTIVEFILE=/var/news/lib/active \
 	 RSHPATH=/usr/bin/rsh \
 	 BASECFLAGS="-g" \
-	 BASELDFLAGS="-lsocket -lnsl -lgen" \
+	 BASELDFLAGS="../c-client/c-client.a -lsocket -lnsl -lgen" \
 	 RANLIB=true
 
 vul:	# VAX Ultrix
 	$(BUILD) `$(CAT) SPECIALS` OS=ult \
 	 SIGTYPE=psx CHECKPW=ult CRXTYPE=nfs \
 	 BASECFLAGS="-O2 -Dconst=" \
-	 BASELDFLAGS="-lauth -lc"
+	 BASELDFLAGS="../c-client/c-client.a -lauth -lc"
 
 vu2:	# VAX Ultrix 2.3, etc.
 	$(BUILD) `$(CAT) SPECIALS` OS=$@ \
 	 CRXTYPE=nfs \
+	 BASELDFLAGS="../c-client/c-client.a" \
 	 BASECFLAGS="-O2 -Dconst= -Dvoid=char"
 
 
@@ -870,7 +907,7 @@ $(ARCHIVE): $(BINARIES)
 # Cleanup
 
 clean:
-	sh -c '$(RM) auths.c crexcl.c ip_unix.c linkage.[ch] siglocal.c osdep*.[ch] *.o ARCHIVE *FLAGS *TYPE $(ARCHIVE) || true'
+	sh -c '$(RM) auths.c crexcl.c ip_unix.c linkage.[ch] siglocal.c osdep*.[ch] *.o ARCHIVE ARCHIVENAME *FLAGS *TYPE $(ARCHIVE) || true'
 
 
 # Dependencies
@@ -979,6 +1016,7 @@ onceenv:
 	echo $(BASELDFLAGS) $(EXTRALDFLAGS) > LDFLAGS
 	echo "$(ARRC) $(ARCHIVE) $(BINARIES);$(RANLIB) $(ARCHIVE)" > ARCHIVE
 	echo $(OS) > OSTYPE
+	echo $(ARCHIVE) >ARCHIVENAME
 	./drivers $(EXTRADRIVERS) $(DEFAULTDRIVERS) dummy
 	./mkauths $(EXTRAAUTHENTICATORS) $(DEFAULTAUTHENTICATORS)
 	echo "  mail_versioncheck (CCLIENTVERSION);" >> linkage.c
diff -urNp -x '*.orig' imap-2007f.org/src/tmail/Makefile imap-2007f/src/tmail/Makefile
--- imap-2007f.org/src/tmail/Makefile	2011-07-23 02:19:58.000000000 +0200
+++ imap-2007f/src/tmail/Makefile	2021-09-29 21:57:46.362676816 +0200
@@ -26,14 +26,14 @@
 
 
 C = ../c-client
-CCLIENTLIB = $C/c-client.a
+CCLIENTLIB = $C/`cat $C/ARCHIVENAME`
 SHELL = /bin/sh
 
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
 CFLAGS = -I$C `cat $C/CFLAGS`
-LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
+LDFLAGS = `cat $C/LDFLAGS`
 
 tmail: $(CCLIENTLIB) tmail.o tquota.o
 	$(CC) $(CFLAGS) -o tmail tmail.o tquota.o $(LDFLAGS)
